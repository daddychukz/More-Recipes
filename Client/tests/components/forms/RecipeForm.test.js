import React from 'react';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ConnectedRecipeForm,
{ PureRecipeForm } from '../../../src/components/forms/RecipeForm';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  recipeOperations: {
    pagination: {},
    recipes: []
  }
};
const store = mockStore(initialState);

const props = {
  addRecipe: sinon.spy(() => new Promise(() => {}))
};

const mountedWrapper = mount(
  <BrowserRouter>
    <ConnectedRecipeForm store={store} />
  </BrowserRouter>
);

const shallowWrapper = shallow(<PureRecipeForm {...props} />);
const state = {
  Title: '',
  Description: '',
  ImageUrl: 'https://res.cloudinary.com/chuks-andela32/image/upload/v1509088084/home_gipmmy.jpg',
  PublicId: 'home_gipmmy.jpg',
  error: false,
};
const event = {
  preventDefault: jest.fn(),
  target: {
    Title: 'Yam',
    Description: 'This is how it is prepared',
  }
};

describe('Add recipe test actions', () => {
  it('renders Add recipe component without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('RecipeForm').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <PureRecipeForm {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls uploadWidget function when user adds image successfully', () => {
    sinon.spy(shallowWrapper.instance(), 'uploadWidget');
    shallowWrapper.instance().uploadWidget();
    expect(shallowWrapper.instance().uploadWidget.calledOnce).toEqual(true);
  });
});
