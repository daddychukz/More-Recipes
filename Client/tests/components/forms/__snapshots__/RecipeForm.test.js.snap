// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add recipe test actions should match component snapshot 1`] = `
<div>
  <form
    id="recipe-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="text-dark"
        htmlFor="title"
      >
        Recipe Title
      </label>
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="text-dark"
        htmlFor="description"
      >
        Recipe Description
      </label>
      <textarea
        className="form-control"
        name="description"
        onChange={[Function]}
        required={true}
        rows="6"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <div>
        <img
          src="http://res.cloudinary.com/chuks-andela32/image/upload/c_scale,dpr_auto,h_150,w_200/home_gipmmy.jpg"
        />
      </div>
      <input
        className="form-control-file"
        onClick={[Function]}
        type="button"
        value="Add Image"
      />
      <small
        className="form-text text-muted"
        id="fileHelp"
      >
        Please upload an example image of recipe for better reviews.
      </small>
    </div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Post Recipe
    </button>
  </form>
</div>
`;
